### PR TITLE
Remove ListingTable and FileScanConfig Unbounded (#8540)

### LINKED ISSUE #8540: Remove ListingTable and FileScanConfig Unbounded
### Is your feature request related to a problem or challenge?

Following on from #7994 it should be possible to remove the unbounded stream support from ListingTable, as this is now provided by StreamTable.

@ozankabak @metesynnada Am I correct in thinking this?

### Describe the solution you'd like

I would like to remove `FileScanConfig::infinite_source` and the corresponding options on CsvReadOptions, etc...

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tustvold**
## Which issue does this PR close?

Closes #8540

### CODE REVIEW

File: datafusion/core/src/physical_optimizer/enforce_sorting.rs [hunk 1]
```diff
@@ -2140,70 +2140,11 @@ mod tests {
         Ok(())
     }
 
-    #[tokio::test]
-    async fn test_with_lost_ordering_unbounded() -> Result<()> {
```
Comment (tustvold) [inline datafusion/core/src/physical_optimizer/enforce_sorting.rs:2144]: I couldn't see a reason to keep these as they effectively duplicate the above

File: datafusion/core/src/physical_optimizer/replace_with_order_preserving_variants.rs [hunk 2]
```diff
@@ -362,23 +362,23 @@ mod tests {
...
         ];
         let expected_optimized = [
             "SortPreservingMergeExec: [a@0 ASC NULLS LAST]",
             "  RepartitionExec: partitioning=Hash([c@1], 8), input_partitions=8, preserve_order=true, sort_exprs=a@0 ASC NULLS LAST",
             "    RepartitionExec: partitioning=RoundRobinBatch(8), input_partitions=1",
-            "      CsvExec: file_groups={1 group: [[file_path]]}, projection=[a, c, d], infinite_source=true, output_ordering=[a@0 ASC NULLS LAST], has_header=true",
+            "      CsvExec: file_groups={1 group: [[file_path]]}, projection=[a, c, d], output_ordering=[a@0 ASC NULLS LAST], has_header=true",
         ];
-        assert_optimized!(expected_input, expected_optimized, physical_plan);
+        assert_optimized!(expected_input, expected_optimized, physical_plan, true);
```
Comment (tustvold) [inline datafusion/core/src/physical_optimizer/replace_with_order_preserving_variants.rs:373]: #8572

File: datafusion/core/src/physical_optimizer/replace_with_order_preserving_variants.rs
Comment (metesynnada) [inline datafusion/core/src/physical_optimizer/replace_with_order_preserving_variants.rs:1]: I think test logic may depend on infinite sources. So, we can use an unbounded executor like `StreamingTableExec` to preserve the test logic.

File: datafusion/core/src/physical_optimizer/test_utils.rs [hunk 3]
```diff
@@ -54,14 +55,19 @@ async fn register_current_csv(
...
-        table_name,
-        &format!("{testdata}/csv/aggregate_test_100.csv"),
-        CsvReadOptions::new()
-            .schema(&schema)
-            .mark_infinite(infinite),
-    )
-    .await?;
+    let path = format!("{testdata}/csv/aggregate_test_100.csv");
+
+    match infinite {
```
Comment (metesynnada) [inline datafusion/core/src/physical_optimizer/test_utils.rs:60]: 👍🏻

**Review by metesynnada — COMMENTED** [review]
Overall LGTM. Just a minor issue on tests.

File: datafusion/core/src/physical_optimizer/replace_with_order_preserving_variants.rs
Comment (tustvold) [reply datafusion/core/src/physical_optimizer/replace_with_order_preserving_variants.rs:1]: It needs either infinite sources or prefer_existing_order, I switched it to the latter

File: datafusion/core/src/physical_optimizer/enforce_sorting.rs (hunk 1 above)
Comment (mustafasrepo) [reply datafusion/core/src/physical_optimizer/enforce_sorting.rs:2144]: These tests makes sure that when sources are unbounded, we try to preserve existing ordering as much as possible. I will re-add corresponding tests that enforces this functionality. For this PR, I just ignore these tests.

**Review by mustafasrepo — APPROVED** [review]
LGTM!. Thanks @tustvold for this cleanup.

**Pull request merged**